pkg/controller.v1beta1/experiment/manifest/generator_test.go

```diff
@@ -244,19 +217,22 @@ spec:
             - "--momentum=0.9"`
 
 	expectedRunSpec, err := util.ConvertStringToUnstructured(expectedStr)
-	if err != nil {
-		t.Errorf("ConvertStringToUnstructured failed: %v", err)
+	if diff := cmp.Diff(nil, err, cmpopts.EquateErrors()); len(diff) != 0 {
+		t.Errorf("ConvertStringToUnstructured failed (-want,+got):\n%s", diff)
 	}
 
-	tcs := []struct {
-		instance             *experimentsv1beta1.Experiment
-		parameterAssignments []commonapiv1beta1.ParameterAssignment
-		err                  bool
-		testDescription      string
+	cases := map[string]struct {
+		prepareConfigMap               func()
+		instance                       *experimentsv1beta1.Experiment
+		parameterAssignments           []commonapiv1beta1.ParameterAssignment
+		wantRunSpecWithHyperParameters *unstructured.Unstructured
+		wantError                      error
 	}{
-		// Valid run
-		// validGetConfigMap1 case
-		{
+		"Run with valid parameters": {
+			prepareConfigMap: func() {
+				c.EXPECT().GetConfigMap(gomock.Any(), gomock.Any()).Return(
+					map[string]string{templatePath: trialSpec}, nil)
+			},
```

Comment:
TBH, I think that we should use the client-go fake client instead of go mock.
But, that is another problem. So, could you open an issue to replace go mock with fake client-go?

Reply:
Done.